tests/Integration/Routing/Fixtures/app.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Illuminate\Tests\Integration\Routing\Fixtures;
+
+use Illuminate\Foundation\Application;
+
+if (! class_exists(AppCache::class)) {
+    class AppCache
+    {
+        public static $app;
+    }
+}
+
+if (isset($refresh)) {
+    return AppCache::$app = Application::configure(basePath: __DIR__)->create();
+} else {
+    return AppCache::$app ??= Application::configure(basePath: __DIR__)->create();
+}
```

Comment:
This is a little weird; I know.

When routes are cached, a new application instance is resolved.

This file allows us to capture the app instance created to add routes to it.